src/main/java/com/google/api/codegen/SingleInterfaceView.java

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen;
+
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+
+import java.util.ArrayList;
+
+/**
+ * A view of the model, that presents only the first reachable interface
+ */
+public class SingleInterfaceView implements InputElementView<Interface> {
+
+  /**
+   * Gets the first reachable interface of the model.
+   */
+  @Override
+  public Iterable<Interface> getElementIterable(Model model) {
```

Comment:
It feels like we are contorting ourselves to fit the existing JavaProvider, which assumes that its input element is an interface, but in reality we are operating over multiple interfaces. Our snippet then has to navigate our object structure to get access to all of the interfaces. The real entity we should be operating on for this package-info generation is some level above the interface.